File: src/main/java/discretemath/bool/expression/atomic/BooleanVariables.java
package discretemath.bool.expression.atomic;

import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

import discretemath.common.VariableSymbols;

public class BooleanVariables {

	public static final BooleanVariable X = new BooleanVariable('x');
	public static final BooleanVariable Y = new BooleanVariable('y');
	public static final BooleanVariable Z = new BooleanVariable('z');

	public static List<BooleanVariable> getVariables(int variablesCount) {
		return StreamSupport.stream(VariableSymbols.getIterator(variablesCount).spliterator(), false)
				.map(BooleanVariable::new)
				.collect(Collectors.toList());
	}
}

File: src/main/scala/discretemath/bool/expression/atomic/BooleanVariables.scala
package discretemath.bool.expression.atomic;

import discretemath.common.VariableSymbols;

object BooleanVariables {

  val X: BooleanVariable = BooleanVariable('x')
  val Y: BooleanVariable = BooleanVariable('y')
  val Z: BooleanVariable = BooleanVariable('z')

  def getVariables(variablesCount: Int): IndexedSeq[BooleanVariable] = {
    return VariableSymbols.view.take(variablesCount).map(symbol => BooleanVariable(symbol)).toIndexedSeq
  }
}
